src/web/components/SlotBodyEnd/ReaderRevenueEpic.tsx

```diff
@@ -108,25 +128,72 @@ const buildPayload = async (props: Props) => {
 			mvtId: Number(getCookie('GU_mvt_id')),
 			countryCode: props.countryCode,
 		},
-	};
+	} as Metadata; // Metadata type incorrectly does not include required hasOptedOutOfArticleCount property
 };
 
-const MemoisedInner = ({
+export const canShow = ({
 	isSignedIn,
 	countryCode,
 	contentType,
 	sectionName,
 	shouldHideReaderRevenue,
 	isMinuteArticle,
 	isPaidContent,
-	isSensitive,
 	tags,
 	contributionsServiceUrl,
-}: Props) => {
-	const [Epic, setEpic] = useState<React.FC>();
-	const [epicProps, setEpicProps] = useState<{}>();
-	const [epicMeta, setEpicMeta] = useState<TestMeta>();
+}: Props): Promise<CanShowResult> => {
+	if (shouldHideReaderRevenue || isPaidContent) {
+		// We never serve Reader Revenue epics in this case
+		return Promise.resolve({ result: false });
+	}
+	const dataPerf = initPerf('contributions-epic-data');
+	dataPerf.start();
+
+	const forcedVariant = getForcedVariant('epic');
+	const queryString = forcedVariant ? `?force=${forcedVariant}` : '';
+
+	return buildPayload({
+		isSignedIn,
+		countryCode,
+		contentType,
+		sectionName,
+		shouldHideReaderRevenue,
+		isMinuteArticle,
+		isPaidContent,
+		tags,
+		contributionsServiceUrl,
+	})
+		.then((contributionsPayload) =>
+			getBodyEnd(
+				contributionsPayload,
+				`${contributionsServiceUrl}/epic${queryString}`,
+			),
+		)
+		.then((response) => {
+			dataPerf.end();
+			return checkForErrors(response);
+		})
+		.then((response) => response.json())
+		.then((json: { data: PreEpicConfig }) => {
```

Comment:
So `PreEpicConfig` describes what we get back from the support-dotcom-components service? I think when the contributions service has nothing to show an empty object is returned (also suggested by the `json.data` check a couple of lines below), so I think data should be optional here:

```js
.then((json: { data?: PreEpicConfig }) => {
```

Reply:
Ah great shout - I'll make that change.